.github/workflows/cleanEnvironments.js

```diff
@@ -47,73 +46,61 @@ program
+    return environment.domain.match(environmentsToCleanRegex);
   });
   return environmentsForReleaseTesting;
 }
 
-async function getEnvironments(token) {
-  const response = await fetch(
-    `${pathToCloudAPI}organizations/saleor/environments/`,
-    {
-      method: "GET",
-      headers: {
-        Authorization: `Token ${token}`,
-        Accept: "application/json",
-        "Content-Type": "application/json;charset=UTF-8",
-      },
-    },
-  );
-  return await response.json();
+async function getEnvironments() {
+  const result = await exec("npx saleor env list --json", {
+    serialization: "json",
+  });
+  const envList = result.stdout.replace("\x1B[?25l\x1B[?25h", "");
+  return JSON.parse(envList);
 }
 
-async function cleanEnvironment(environment, snapshot, token) {
-  const response = await fetch(
-    `${pathToCloudAPI}organizations/saleor/environments/${environment.key}/restore/`,
+async function cleanEnvironment(environment, snapshot) {
+  console.log(`Restoring snapshot for environment ${environment.domain}`);
+  const result = await exec(
+    `npx saleor backup restore ${snapshot.key} --environment="${environment.key}" --skip-webhooks-update`,
     {
-      method: "PUT",
-      body: JSON.stringify({ restore_from: snapshot.key }),
-      headers: {
-        Authorization: `Token ${token}`,
-        Accept: "application/json",
-        "Content-Type": "application/json;charset=UTF-8",
-      },
+      serialization: "json",
     },
   );
-  const responseInJson = await response.json();
-  if (
-    responseInJson.non_field_errors
-      ? responseInJson.non_field_errors
-      : responseInJson.__all__
-  ) {
-    const warning = responseInJson.non_field_errors
-      ? responseInJson.non_field_errors
-      : responseInJson.__all__;
+  const response = result.stderr.replace("\x1B[?25l\x1B[?25h", "");
+
+  if (!response.includes("Restore finished")) {
+    const warning = response;
     console.warn(`${environment.name}: ${warning}`);
     sendWarningOnSlack = "true";
     warningMessage += `Could not revert snapshot on ${environment.domain}: ${warning}.\n`;
-  } else {
-    await waitUntilTaskInProgress(responseInJson.task_id, environment.name);
-  }
+  } 
 }
 
-async function getSnapshotsForRestore(token) {
-  const snapshotsResponse = await fetch(
-    `${pathToCloudAPI}organizations/saleor/backups/`,
-    {
-      method: "GET",
-      headers: {
-        Authorization: `Token ${token}`,
-        Accept: "application/json",
-        "Content-Type": "application/json;charset=UTF-8",
-      },
-    },
+async function getSnapshotsForRestore(allEnvironments) {
+  let allBackups = [];
+  await Promise.all(
+    allEnvironments.map(async environment => {
+      const result = await exec(
+        `npx saleor backup list --json --organization="saleor" --environment="${environment.key}"`,
+        {
+          serialization: "json",
+        },
+      );
```

Comment:
Use `child_process.execFile(file[, args][, options][, callback])` instead, `child_process.exec(command[, options][, callback])` is not designed for handling unsanitized arguments